webapp/src/reducers.ts

```diff
@@ -34,16 +32,25 @@ import {
     RECEIVED_CALLS_CONFIG,
     SHOW_END_CALL_MODAL,
     HIDE_END_CALL_MODAL,
+    RECEIVED_CHANNEL_STATE,
 } from './action_types';
 
-const isVoiceEnabled = (state = false, action: { type: string }) => {
+interface channelState {
+    id: string,
+    enabled: boolean,
+}
+
+interface channelStateAction {
+    type: string,
+    data: channelState,
+}
+
+const channelState = (state: {[channelID: string]: channelState} = {}, action: channelStateAction) => {
```

Comment:
An alternative would be:
```suggestion
const channelState = (state: Record<string, channelState> = {}, action: channelStateAction) => {
```
But I kind of like that we're saying the key is a `channelID`... 👍

Reply:
What do you think of deferring this type of change to a dedicated ticket where we consistently make use of it?

Reply:
Sounds good. But to be clear: I think using `{[channelID: string]: channelState}` is fine, and maybe even better than `Record` because it tells us the index is a channelID.